third_party/move/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -255,6 +255,98 @@ impl<'env> SpecTranslator<'env> {
         }
     }
 
+    /// Generates axioms for uninterpreted spec function.
```

Comment:
As I said before, currently all `fun.spec` are eliminated before we hit the boogie backend. All the logic for this is in a different [`spec_translator`](https://github.com/aptos-labs/aptos-core/blob/d6acef0e46e6d5c036bb7ebd669083a74ab9caf4/third_party/move/move-model/src/spec_translator.rs#L201), which is called from the [`spec_instrumentation`](https://github.com/aptos-labs/aptos-core/blob/97a555b0d13faa1f498be4debeafd646664e9e45/third_party/move/move-prover/bytecode-pipeline/src/spec_instrumentation.rs#L237). At the moment we reach the boogie backend, they are only asserts and assumes left. 

Feels like the axioms should also be generated on this level (spec_instrumentation?) and not in the Boogie backend.

Reply:
I tend to keep generation of axioms at the boogie level at least for now because they need to refer to the result of the spec function call, which is much easier to directly generate the boogie code, following emitting axiom about return type for the spec function.